File: demo_android/EmptyLoading/library/src/main/java/better/android/library/empty/EmptyViewRefreshProxy.java
package better.android.library.empty;

import android.view.View;

public interface EmptyViewRefreshProxy {

    void displayLoading();

    void displayLoading(int msgId);

    /**
     * 显示请求等待。
     *
     * @param msg 等待提示消息。
     */
    void displayLoading(String msg);

    void displayMessage();

    void displayMessage(int msg);

    /**
     * 请求成功 当列表为空的时候展示的信息
     * Create By better on 2017/4/2 07:14.
     */
    void displayMessage(String msg);

    void displayRetry();

    void displayRetry(int msgId);

    /**
     * 请求失败 显示请求重试。
     *
     * @param msg 重试提示消息：失败、无数据。
     */
    void displayRetry(String msg);

    /**
     * 直接消失emptyView
     */

    void disappear();

    View getProxyView();

    void setOnRetryClickListener(OnLrpRetryClickListener listener);
}

File: demo_android/EmptyLoading2/libraryEmpty/src/main/java/com/better/library/empty/EmptyViewRefreshProxy.kt
package com.better.library.empty

import android.view.View

interface EmptyViewRefreshProxy {

    val proxyView: View

    fun displayLoading(msg: Int)

    /**
     * 显示请求等待。
     *
     * @param msg 等待提示消息。
     */
    fun displayLoading(msg: String = proxyView.context.getString(R.string.empty_tip_loading))

    fun displayMessage(msg: Int)

    /**
     * 请求成功 当列表为空的时候展示的信息
     * Create By better on 2017/4/2 07:14.
     */
    fun displayMessage(msg: String = proxyView.context.getString(R.string.empty_tip_empty))

    fun displayRetry(msg: Int)

    /**
     * 请求失败 显示请求重试。
     *
     * @param msg 重试提示消息：失败、无数据。
     */
    fun displayRetry(msg: String = proxyView.context.getString(R.string.empty_tip_fail))

    /**
     * 直接消失emptyView
     */
    fun disappear()

    fun setOnRetryClickListener(listener: OnLrpRetryClickListener)
}
